File: backend/app/schemas/cash.py
"""Cash payment logging & end-of-day reconciliation schemas (Section 11)."""

from __future__ import annotations

from datetime import date

from pydantic import BaseModel

from app.core.constants import PaymentMethod


class CashPaymentLog(BaseModel):
    """Front-desk logs an offline payment for a member (Section 11.1)."""

    member_id: str
    plan_id: str
    amount: float
    method: PaymentMethod = PaymentMethod.CASH
    note: str | None = None


class CashPaymentOut(BaseModel):
    payment_id: str
    member_id: str
    amount: float
    method: str
    member_status: str
    receipt_pdf_url: str | None = None


class ReconciliationRequest(BaseModel):
    business_date: date
    counted_total: float
    notes: str | None = None


class ReconciliationOut(BaseModel):
    id: str
    business_date: date
    system_total: float
    counted_total: float
    discrepancy: float
    performed_by: str
    alert_triggered: bool = False

File: shared/types/cash.ts
export interface CashPayment {
  payment_id: string;
  organization_id: string;
  member_id: string;
  member_email: string;
  member_name: string | null;
  plan_id: string | null;
  amount: number;
  method: string;
  logged_by: string;
  logged_by_name: string | null;
  member_status: string;
  receipt_pdf_url: string | null;
  created_at: string;
}

export interface CashReconciliation {
  id: string;
  organization_id: string;
  business_date: string;
  system_total: number;
  counted_total: number;
  discrepancy: number;
  performed_by: string;
  performed_by_name: string | null;
  alert_triggered: boolean;
  notes: string | null;
  created_at: string;
}

export interface CashLogEntry {
  id: string;
  organization_id: string;
  member_id: string;
  amount: number;
  method: string;
  note: string | null;
  staff_member_id: string;
  created_at: string;
}
